reyna/src/com/b2msolutions/reyna/system/Preferences.java

```diff
@@ -196,4 +199,25 @@ private String getString(String key, String defaultValue) {
         SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
         return sp.getString(key, defaultValue);
     }
+
+    public long getNonRecurringWwanBlackoutStartTime() {
+        return this.getLong(WWAN_BLACKOUT_START, -1);
```

Comment:
Solution to this could be saving the utcs as strings and then when we use them parse them to long. I think I will do that.